src/conf/duration.rs

```diff
@@ -0,0 +1,126 @@
+/// Provides deserialize [`time::Duration`] from string.
+use std::fmt;
+use std::time::Duration;
+
+use serde::de::{self, Unexpected};
+use serde::Serializer;
+
+struct DurationFromStringVisitor;
+
+pub fn deserialize<'de, D>(d: D) -> Result<Duration, D::Error>
+where
+    D: de::Deserializer<'de>,
+{
+    Ok(d.deserialize_str(DurationFromStringVisitor)?)
+}
+
+impl<'de> de::Visitor<'de> for DurationFromStringVisitor {
+    type Value = Duration;
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        write!(formatter, "a string representation of duration")
+    }
+
+    fn visit_str<E>(self, value: &str) -> Result<Duration, E>
+    where
+        E: de::Error,
+    {
+        let dur = humantime::parse_duration(value).map_err(|_| {
+            de::Error::invalid_value(Unexpected::Str(value), &self)
```

Comment:
Wow, this seems tough. This should be implemented in some `util` module, or even a separate crate.

Fortunately, reading [documentation carefully](https://github.com/tailhook/humantime/blob/7291944f4a626e00b8c57f678242532395035b22/README.md#license) allows to avoid writing redudant code: [serde-humantime](https://docs.rs/serde-humantime) 🙃

Reply:
@tyranron ,

Прикрутил, но там только `deserialize`. А `serialize` используется в [этом](https://github.com/instrumentisto/medea/blob/463c6a94a76dff08d6abce1f965a900942ed1164/src/conf/mod.rs#L37) финте, который там не от хорошей жизни. Получится выкосить его - выкошу `serialize`.

Reply:
@alexlapa I think we should anyway have `Serialize` capability, even for debug purposes, like "export me the current config".

A bit more exploring and we have [this](https://github.com/sfackler/serde-humantime/pull/3#issuecomment-425640627).